### PR TITLE
Update docs for new clause count setting

### LINKED ISSUE #7385: [DOC]Change the docs around `max_clause_count`
**What do you want to do?**
 
- [X] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 We're making a change to `max_clause_count` from being a static setting to be dynamically update-able. We would need to update the documentation to reflect this change.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
https://github.com/opensearch-project/OpenSearch/pull/13568

### PR DESCRIPTION
**Pull request by harshavamsi**
### Description
Adds documentation for new clause count setting

### Issues Resolved
Closes #7385 

### Version
2.16

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 1]
```diff
@@ -39,6 +39,8 @@ OpenSearch supports the following cluster-level index settings. All settings in
...
 
+- `indices.query.bool.max_clause_count` (Integer): Defines the maximum value for the product of the number of fields and the number of terms that can be queried at one time. Default value is 1024. Note that prior to OpenSearch versions 2.16, this was a static setting that required a cluster restart to take effect. Due to the nature of changing a static setting to being dynamic, it is possible that search threadpools hold the older static value before the dynamic setting is applied on all nodes thereby causing certain search requests to still error out with `TooManyClauses` exception. When new search threadpools are created, they will hold a reference to the new value.
```
Comment (vagimeli) [inline _install-and-configure/configuring-opensearch/index-settings.md:42]:
```suggestion
- `indices.query.bool.max_clause_count` (Integer): Defines the maximum product of fields and terms queryable at once. Before OpenSearch 2.16, this static setting required a cluster restart to apply. Now dynamic, search thread pools may retain the old static value initially, causing TooManyClauses exceptions. New threadpools use the updated value. Default is 1024.
```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 2]
```diff
@@ -39,6 +39,8 @@ OpenSearch supports the following cluster-level index settings. All settings in
 
 - `indices.fielddata.cache.size` (String): The maximum size of the field data cache. May be specified as an absolute value (for example, `8GB`) or a percentage of the node heap (for example, `50%`). This value is static so you must specify it in the `opensearch.yml` file. If you don't specify this setting, the maximum size is unlimited. This value should be smaller than the `indices.breaker.fielddata.limit`. For more information, see [Field data circuit breaker]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/circuit-breaker/#field-data-circuit-breaker-settings).
 
+- `indices.query.bool.max_clause_count` (Integer): Defines the maximum product of fields and terms queryable at once. Before OpenSearch 2.16, this static setting required a cluster restart to apply. Now dynamic, existing search thread pools may use the old static value initially, causing `TooManyClauses` exceptions. New thread pools use the updated value. Default is 1024.
```
Comment (vagimeli) [inline _install-and-configure/configuring-opensearch/index-settings.md:42]:
```suggestion
- `indices.query.bool.max_clause_count` (Integer): Defines the maximum product of fields and terms queryable at once. Before OpenSearch 2.16, this static setting required a cluster restart to apply. Now dynamic, existing search thread pools may use the old static value initially, causing `TooManyClauses` exceptions. New thread pools use the updated value. Default is `1024`.
```
```suggestion
- `indices.query.bool.max_clause_count` (Integer): Defines the maximum product of fields and terms queryable at once. Before OpenSearch 2.16, this static setting required a cluster restart to apply. Now dynamic, existing search thread pools may use the old static value initially, causing `TooManyClauses` exceptions. New thread pools use the updated value. Default is 1024.
```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 3]
```diff
@@ -39,6 +39,8 @@ OpenSearch supports the following cluster-level index settings. All settings in
 
 - `indices.fielddata.cache.size` (String): The maximum size of the field data cache. May be specified as an absolute value (for example, `8GB`) or a percentage of the node heap (for example, `50%`). This value is static so you must specify it in the `opensearch.yml` file. If you don't specify this setting, the maximum size is unlimited. This value should be smaller than the `indices.breaker.fielddata.limit`. For more information, see [Field data circuit breaker]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/circuit-breaker/#field-data-circuit-breaker-settings).
 
+- `indices.query.bool.max_clause_count` (Integer): Defines the maximum product of fields and terms queryable at once. Before OpenSearch 2.16, this static setting required a cluster restart to apply. Now dynamic, existing search thread pools may use the old static value initially, causing `TooManyClauses` exceptions. New thread pools use the updated value. Default is `1024`.
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/index-settings.md:42]:
```suggestion
- `indices.query.bool.max_clause_count` (Integer): Defines the maximum product of fields and terms that are queryable at once. Before OpenSearch 2.16, a cluster restart was required in order to apply this static setting. Now dynamic, existing search thread pools may use the old static value initially, causing `TooManyClauses` exceptions. New thread pools use the updated value. Default is `1024`.
```

Comment (natebower) [inline _install-and-configure/configuring-opensearch/index-settings.md:42]: "at once" => "simultaneously"?

**Review by natebower — COMMENTED** [review]
@vagimeli @harshavamsi Just a few changes. Thanks!

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 4]
```diff
@@ -39,6 +39,8 @@ OpenSearch supports the following cluster-level index settings. All settings in
 
 - `indices.fielddata.cache.size` (String): The maximum size of the field data cache. May be specified as an absolute value (for example, `8GB`) or a percentage of the node heap (for example, `50%`). This value is static so you must specify it in the `opensearch.yml` file. If you don't specify this setting, the maximum size is unlimited. This value should be smaller than the `indices.breaker.fielddata.limit`. For more information, see [Field data circuit breaker]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/circuit-breaker/#field-data-circuit-breaker-settings).
 
+- `indices.query.bool.max_clause_count` (Integer): Defines the maximum product of fields and terms that are queryable at once. Before OpenSearch 2.16, a cluster restart was required in order to apply this static setting. Now dynamic, existing search thread pools may use the old static value initially, causing `TooManyClauses` exceptions. New thread pools use the updated value. Default is `1024`.
```
Comment (vagimeli) [inline _install-and-configure/configuring-opensearch/index-settings.md:42]:
```suggestion
- `indices.query.bool.max_clause_count` (Integer): Defines the maximum product of fields and terms that are queryable simultaneously. Before OpenSearch 2.16, a cluster restart was required in order to apply this static setting. Now dynamic, existing search thread pools may use the old static value initially, causing `TooManyClauses` exceptions. New thread pools use the updated value. Default is `1024`.
```

**Review by vagimeli — APPROVED** [review]
Doc and editorial reviews completed

**Pull request merged**